### ISSUE TITLE
Segmentation fault from non target node during nodetool decommission 

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylla#9960]
*Installation details*
Scylla version (or git commit hash):4.7.dev-0.20220118.985403ab9 with build-id d48bc5d3676f53140f58fae39086e98f17a58534
Cluster size: 6
OS (RHEL/CentOS/Ubuntu/AWS AMI):ami-057906d0f869ccd53 (eu-west-1)
Scenario: https://github.com/scylladb/scylla-cluster-tests/blob/master/test-cases/longevity/longevity-cdc-100gb-4h.yaml
### This scenario uses CDC

At `2022-01-23 05:12:43,155` a nodetool decomission has started on node 6 (34.246.223.186 | 10.0.1.42). During the decommission, however, at `2022-01-23T05:16:34+00`, node 5 (3.250.67.35 | 10.0.2.210) experienced a segmentation fault:
```
Segmentation fault on shard 1.
Backtrace:
  0x424fef8
  0x4280796
  0x48cebd1
  0x7f1328fafa1f
  0x1713852
  0x1716edb
  0x4260e54
  0x4262227
  0x4280d05
  0x42341ba
  /opt/scylladb/libreloc/libpthread.so.0+0x9298
  /opt/scylladb/libreloc/libc.so.6+0x100352
```
Coredump:
```
backtrace=           PID: 15880 (scylla)
           UID: 113 (scylla)
           GID: 119 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Sun 2022-01-23 05:16:34 UTC (4min 25s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1 --experimental-features cdc --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15 --lock-memory=1
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: e957234bcc684e2f93d8263470ab3a64
    Machine ID: 122c2e3e25124b2dbcf5ddb2c894c926
      Hostname: longevity-cdc-100gb-4h-master-db-node-7fdcde8d-5
       Storage: /var/lib/systemd/coredump/core.scylla.113.e957234bcc684e2f93d8263470ab3a64.15880.1642914994000000000000
       Message: Process 15880 (scylla) of user 113 dumped core.
                
                Stack trace of thread 15881:
                #0  0x0000000001713853 _ZN7seastar8futurizeINS_6futureIvEEE6invokeIRZN12_GLOBAL__N_112shard_reader5closeEvE4$_32JEEES2_OT_DpOT0_ (scylla + 0x1513853)
                #1  0x0000000001716edc _ZN7seastar17smp_message_queue15async_work_itemIZN12_GLOBAL__N_112shard_reader5closeEvE4$_32E15run_and_disposeEv (scylla + 0x1516edc)
                #2  0x0000000004260e55 _ZN7seastar7reactor14run_some_tasksEv (scylla + 0x4060e55)
                #3  0x0000000004262228 _ZN7seastar7reactor6do_runEv (scylla + 0x4062228)
                #4  0x0000000004280d06 _ZNSt17_Function_handlerIFvvEZN7seastar3smp9configureERKNS1_11smp_optionsERKNS1_15reactor_optionsEE5$_100E9_M_invokeERKSt9_Any_data (scylla + 0x4080d06)
                #5  0x00000000042341bb _ZN7seastar12posix_thread13start_routineEPv (scylla + 0x40341bb)
                #6  0x00007f1328fa5299 start_thread (libpthread.so.0 + 0x9299)
                #7  0x00007f13284b7353 __clone (libc.so.6 + 0x100353)
...
download_instructions=gsutil cp gs://upload.scylladb.com/core.scylla.113.e957234bcc684e2f93d8263470ab3a64.15880.1642914994000000000000/core.scylla.113.e957234bcc684e2f93d8263470ab3a64.15880.1642914994000000000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.113.e957234bcc684e2f93d8263470ab3a64.15880.1642914994000000000000.gz
```

The decommission process ended failing:
```
Command: '/usr/bin/nodetool  decommission '

Exit code: 1

Stdout:

nodetool: Scylla API server HTTP POST to URL '/storage_service/decommission' failed: std::runtime_error (node_ops_cmd_check: Node 10.0.1.42 rejected node_ops_cmd=decommission_abort from node=10.0.1.42 with ops_uuid=700c8e87-d6c9-4f3a-bba6-8ad496520802, pending_node_ops={}, the node ops is unknown)
See 'nodetool help' or 'nodetool help <command>'.

Stderr:


```
After the decommission ended abruptly, node 6 was removed from the cluster:
```
< t:2022-01-23 05:25:31,450 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > Datacenter: eu-west
< t:2022-01-23 05:25:31,467 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > ===================
< t:2022-01-23 05:25:31,467 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > Status=Up/Down
< t:2022-01-23 05:25:31,467 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2022-01-23 05:25:31,483 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > --  Address     Load       Tokens       Owns    Host ID                               Rack
< t:2022-01-23 05:25:31,529 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.210  4.78 GB    256          ?       71fc2cfa-5d7e-4905-a939-fd077e57b7ec  1a
< t:2022-01-23 05:25:31,579 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > UN  10.0.1.87   5.57 GB    256          ?       3b20f5f0-3b22-40d2-8da8-ea54faf618e2  1a
< t:2022-01-23 05:25:31,602 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > UN  10.0.1.184  5.16 GB    256          ?       9ed61012-5456-494b-b649-c1e4825170ce  1a
< t:2022-01-23 05:25:31,625 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.156  5.82 GB    256          ?       ad80779f-3248-4fe5-9d08-0cf50f2ace15  1a
< t:2022-01-23 05:25:31,664 f:base.py         l:222  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.255  5.01 GB    256          ?       d8a34bbc-47e1-48ab-b187-21f9e0fecdb8  1a
```
The node iteslf, however, continued to stay alive.

Logs:
```
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                              Log links for testrun with test id 7fdcde8d-ef9e-4c8f-ab4a-9032486f4689                                                                                                              |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type       | Link                                                                                                                                                                                                                                                           |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20220123_081036 | grafana        | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_081036/grafana-screenshot-longevity-cdc-100gb-4h-test-scylla-per-server-metrics-nemesis-20220123_081036-longevity-cdc-100gb-4h-master-monitor-node-7fdcde8d-1.png |
| 20220123_082027 | critical       | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/critical-7fdcde8d.log.tar.gz                                                                                                                               |
| 20220123_082027 | db-cluster     | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/db-cluster-7fdcde8d.tar.gz                                                                                                                                 |
| 20220123_082027 | debug          | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/debug-7fdcde8d.log.tar.gz                                                                                                                                  |
| 20220123_082027 | email_data     | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/email_data-7fdcde8d.json.tar.gz                                                                                                                            |
| 20220123_082027 | error          | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/error-7fdcde8d.log.tar.gz                                                                                                                                  |
| 20220123_082027 | event          | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/events-7fdcde8d.log.tar.gz                                                                                                                                 |
| 20220123_082027 | left_processes | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/left_processes-7fdcde8d.log.tar.gz                                                                                                                         |
| 20220123_082027 | loader-set     | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/loader-set-7fdcde8d.tar.gz                                                                                                                                 |
| 20220123_082027 | monitor-set    | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/monitor-set-7fdcde8d.tar.gz                                                                                                                                |
| 20220123_082027 | normal         | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/normal-7fdcde8d.log.tar.gz                                                                                                                                 |
| 20220123_082027 | output         | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/output-7fdcde8d.log.tar.gz                                                                                                                                 |
| 20220123_082027 | event          | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/raw_events-7fdcde8d.log.tar.gz                                                                                                                             |
| 20220123_082027 | sct            | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/sct-7fdcde8d.log.tar.gz                                                                                                                                    |
| 20220123_082027 | summary        | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/summary-7fdcde8d.log.tar.gz                                                                                                                                |
| 20220123_082027 | warning        | https://cloudius-jenkins-test.s3.amazonaws.com/7fdcde8d-ef9e-4c8f-ab4a-9032486f4689/20220123_082027/warning-7fdcde8d.log.tar.gz                                                                                                                                |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by slivne** [comment]
maybe the same as https://github.com/scylladb/scylla/issues/9949

**Comment by slivne** [comment]
@bhalevy maybe we should reassign since botond queue is long :)

**Comment by bhalevy** [comment]
Decoded backtrace:
```
void seastar::backtrace(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
 (inlined by) seastar::backtrace_buffer::append_backtrace() at ./build/release/seastar/./seastar/src/core/reactor.cc:750
 (inlined by) seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:780
seastar::print_with_backtrace(char const*, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:792
 (inlined by) seastar::sigsegv_action() at ./build/release/seastar/./seastar/src/core/reactor.cc:3663
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3649
 (inlined by) __invoke at ./build/release/seastar/./seastar/src/core/reactor.cc:3645
wasmtime_runtime::traphandlers::unix::trap_handler at wasmtime_runtime.9xv9ixev-cgu.15:?
?? ??:0
std::__uniq_ptr_impl >::_M_ptr() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:173
 (inlined by) std::unique_ptr >::get() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:422
 (inlined by) std::unique_ptr >::operator*() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:408
 (inlined by) seastar::foreign_ptr > >::operator*() const at ././seastar/include/seastar/core/sharded.hh:904
 (inlined by) operator() at ./mutation_reader.cc:2110
 (inlined by) seastar::future seastar::futurize >::invoke<(anonymous namespace)::shard_reader::close()::$_32&>((anonymous namespace)::shard_reader::close()::$_32&) at ././seastar/include/seastar/core/future.hh:2135
seastar::smp_message_queue::async_work_item<(anonymous namespace)::shard_reader::close()::$_32>::run_and_dispose() at ././seastar/include/seastar/core/smp.hh:231
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./seastar/src/core/reactor.cc:2344
 (inlined by) seastar::reactor::run_some_tasks() at ./build/release/seastar/./seastar/src/core/reactor.cc:2754
seastar::reactor::do_run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2923
operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:4122
 (inlined by) void std::__invoke_impl(std::__invoke_other, seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_100&) at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/invoke.h:61
 (inlined by) std::enable_if, void>::type std::__invoke_r(seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_100&) at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/invoke.h:111
 (inlined by) std::_Function_handler::_M_invoke(std::_Any_data const&) at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/std_function.h:291
std::function::operator()() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/std_function.h:560
 (inlined by) seastar::posix_thread::start_routine(void*) at ./build/release/seastar/./seastar/src/core/posix.cc:60
```

Duplicate of #9923
Fixed by d7a993043de250f39c3881017b83fbb1e789e6fb
